pennylane/devices/default_mixed.py

```diff
@@ -515,3 +528,9 @@ def apply(self, operations, rotations=None, **kwargs):
         # apply the circuit rotations
         for operation in rotations:
             self._apply_operation(operation)
+
+        # apply the readout error if readout_prob is non-zero
+        if self.readout_err:
+            bitflips = [qml.BitFlip(self.readout_err,wires=k) for k in self.wires]
```

Comment:
@antalszava 
You had a valid point in our discussion that we may not need to apply BitFlip channel to all qubits but only to the ones on which the user requires to do measurements. BitFlip channel on one qubit is a local quantum channel. Applying BitFlip channel to a particular qubit as the last operation in the circuit will not change the quantum state (the reduced density matrix) of any other qubit, and thus will not influence the results of `expval`, `var`, `probs`, and `sample` on other qubits. 

I am wondering how can I extract the list of wires on which any measurement operation has been passed on by the user. Can you please let me know how to do this? Thanks.

Reply:
Hi @meenu-kri, good catch!

We won't be able to extract information about the measured wires easily in the `apply` method. What could be done, is to redefine `statistics` in some way, because the observables are being passed into that method. Something like the following could work (note: this code is not tested):
```py
    def statistics(self, observables, shot_range=None, bin_size=None):
        # apply the readout error if readout_prob is non-zero
        if self.readout_err:
            measured_wires = qml.wires.Wires.all_wires([obs.wires for obs in observables])
            bitflips = [qml.BitFlip(self.readout_err,wires=k) for k in measured_wires]
        return super().statistics(observables, shot_range, bin_size)
```
Also, it will be good to have tests for more of the measurements: `qml.state`, `qml.density_matrix`, `qml.probs`.

Reply:
Hi @antalszava , 

Thanks for the comments! I checked that adding 

```
if self.readout_err:
            measured_wires = qml.wires.Wires.all_wires([obs.wires for obs in observables])
            bitflips = [qml.BitFlip(self.readout_err,wires=k) for k in measured_wires]
```

to `statistics` saves the list of  wires being measured in `measured_wires`. However, I am having trouble in accessing `bitflips` or `measured_wires` variables in the `apply` method in the `DefaultMixed` class.  Can you please comment on this? I tried calling `super().statistics()` in the apply method but it returns an error that statistics is missing the required argument `'observables'`. And we do not have access to `'observables'` in the `apply` method to pass it as an argument.

I will add more tests for  `qml.state`, `qml.density_matrix`, `qml.probs` as you have mentioned.